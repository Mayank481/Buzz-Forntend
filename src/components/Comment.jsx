import React, { useState } from 'react'

export default function Comment({data}) {
    const [loadmore, setloadmore] = useState(true);
    return (
      <>
        <div className="d-flex container bg-light mt-1 align-items-center">
          <div className="">
            {data.picture_url ? (
              <div className="">
                <img
                  data-testid='userCommentPic'
                  src={data.picture_url}
                  className=" mt-2 small-round-pic me-2  round-img "
                  alt="..."
                />
              </div>
            ) : (
              <i className="mt-2 me-2 fa-solid fa-user fa-2x card-img-top small-round-pic  round-img bg-warning d-flex justify-content-center align-items-center"></i>
            )}
          </div>
          <div data-testid='cmnt' className={loadmore && 'text-comment'}>{data.message}</div>
        </div>
        <div
          className="seemore"
          onClick={() => {
            setloadmore((p) => !p);
          }}
        >
          {loadmore ? 'Show More' : 'Show less'}
        </div>
      </>
    );
}
