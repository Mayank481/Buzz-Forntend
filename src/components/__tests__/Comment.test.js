import { render, screen } from '@testing-library/react';
import Comment from '../Comment';
import renderer from 'react-test-renderer';
test('should render Comment Panel', () => {
    const data = {
        picture_url: "no pic",
        message: 'no msg'
    }
    render(<Comment data={data} />)
    const text = screen.queryByTestId('userCommentPic').src;
    const text1 = screen.queryByTestId('cmnt').innerHTML;
    const component = renderer.create(<Comment data={data} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
    expect(text).toBe("http://localhost/no%20pic");
    expect(text1).toBe("no msg");
})
