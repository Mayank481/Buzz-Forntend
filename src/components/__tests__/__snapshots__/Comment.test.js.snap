// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Comment Panel 1`] = `
Array [
  <div
    className="d-flex container bg-light mt-1 align-items-center"
  >
    <div
      className=""
    >
      <div
        className=""
      >
        <img
          alt="..."
          className=" mt-2 small-round-pic me-2  round-img "
          data-testid="userCommentPic"
          src="no pic"
        />
      </div>
    </div>
    <div
      className="text-comment"
      data-testid="cmnt"
    >
      no msg
    </div>
  </div>,
  <div
    className="seemore"
    onClick={[Function]}
  >
    Show More
  </div>,
]
`;
